sections/reading_rust.tex

```diff
@@ -264,24 +264,37 @@ \section{Reading Rust}
 \end{minted}
 \end{frame}
 
+\begin{frame}[fragile]{Trait Implementation}
+\begin{minted}{rust}
+struct Square {
+  x: f64,
+  y: f64,
+  side: f64,
+}
+
+impl HasArea for Square {
+  fn area(&self) -> f64 {
+    self.side * self.side
+  }
+}
+\end{minted}
+\end{frame}
+
 %%%
 
 \begin{frame}[fragile]{Trait Based Generics}
 
 \begin{minted}{rust}
-fn min<T: Ord>(a: T, b: T) -> T {
-  if a <= b { a } else { b }
+fn print_area<Shape: HasArea>(shape: &Shape) {
+  println!("{}", shape.area())
```

Comment:
This method has no return value, so I think the `println!(...)` expression should end with a semi-colon.